service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -438,7 +439,7 @@ public synchronized void init(HiveConf hiveConf) {
     }
 
     // Add a shutdown hook for catching SIGTERM & SIGINT
-    ShutdownHookManager.addShutdownHook(() -> hiveServer2.stop());
+    ShutdownHookManager.addShutdownHook(() -> graceful_stop());
```

Comment:
if the default is graceful_stop() when you CNTRL-C the HS2 process, should we decrease the default timeout from 30 mins to something lower? if HS2 does not have any running queries, I think it would be equivalent to a force shutdown. But regardless, just thinking out loud.
if there is a long running query, how would one issue a force shutdown ?

Reply:
Hi @nrg4878, thank you for the comment.
The `graceful_stop` will check both timeout and number of running queries:
https://github.com/apache/hive/blob/746eba21e39965d56d358ef4c195f64054ef27ec/service/src/java/org/apache/hive/service/server/HiveServer2.java#L928
If there's no running queries, then call `stop` to terminate the process with no delay.
For long running queries(timeout happens), then this case will also call `stop`, the operation will cancel the running queries on stopping:
https://github.com/apache/hive/blob/746eba21e39965d56d358ef4c195f64054ef27ec/service/src/java/org/apache/hive/service/cli/operation/OperationManager.java#L95-L105
Besides the command will force shutdown the process when timeout and the process is still alive:
https://github.com/apache/hive/blob/746eba21e39965d56d358ef4c195f64054ef27ec/bin/ext/hiveserver2.sh#L91-L94